db/write_batch.cc

```diff
@@ -1597,6 +1602,22 @@ class MemTableInserter : public WriteBatch::Handler {
         flush_scheduler_->ScheduleFlush(cfd);
       }
     }
+    // check if memtable_list size exceeds max_write_buffer_size_to_maintain
+    // TODO: need to check if cfd have already been added to
+    // trim_history_scheduler_
+    if (trim_history_scheduler_ != nullptr) {
+      auto* cfd = cf_mems_->current();
+      assert(cfd != nullptr);
+      if (cfd->imm()->imm_trim_needed.load(std::memory_order_relaxed) ==
+              false &&
+          cfd->mem()->ApproximateMemoryUsage() +
+                  cfd->imm()->ApproximateMemoryUsage() >=
+              static_cast<size_t>(
+                  cfd->imm()->current()->GetMaxWriteBufferSizeToMaintain()) &&
+          cfd->imm()->MarkTrimHistoryNeeded()) {
+        trim_history_scheduler_->ScheduleFlush(cfd);
```

Comment:
Consider a better naming than `ScheduleFlush` if you decide to use a separate `trim_history_scheduler`.